src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx

```diff
@@ -118,6 +124,82 @@ describe("Create New Cql Library Component", () => {
     expect(mockPush).toHaveBeenCalledWith("/example");
   });
 
+  it("should render the model options when clicked", async () => {
+    render(<CreateNewCqlLibrary />);
+    const modelDropdown = screen.getByRole("button", {
+      name: /select a model/i,
+    });
+    userEvent.click(modelDropdown);
+    const qiCoreOption = screen.getByText("QI-Core");
+    expect(qiCoreOption).toBeInTheDocument();
+  });
+
+  it("should update the dropdown with the selected option", async () => {
+    render(<CreateNewCqlLibrary />);
+    const modelDropdown = screen.getByRole("button", {
+      name: /select a model/i,
+    });
+    userEvent.click(modelDropdown);
+    const qiCoreOption = screen.getByText("QI-Core");
+    expect(qiCoreOption).toBeInTheDocument();
+    userEvent.click(qiCoreOption);
+    const qiCore = await screen.findByText("QI-Core");
+    expect(qiCore).toBeInTheDocument();
+    const qiCoreButton = screen.getByRole("button", { name: /qi-core/i });
+    expect(qiCoreButton).toBeInTheDocument();
+  });
+
+  it("should display an error when model is not selected", async () => {
+    // jest.setTimeout(15000);
+    render(<CreateNewCqlLibrary />);
+    const input = screen.getByTestId(
+      "cql-library-name-text-field"
+    ) as HTMLInputElement;
+    userEvent.type(input, "TestingLibraryName12");
+    expect(input.value).toBe("TestingLibraryName12");
+    const modelDropdown = screen.getByRole("button", {
+      name: /select a model/i,
+    });
+    expect(modelDropdown).toBeInTheDocument();
+    userEvent.click(modelDropdown);
+    const qiCoreOption = screen.getByText("QI-Core");
+    expect(qiCoreOption).toBeInTheDocument();
+    userEvent.type(modelDropdown, "{esc}");
+    userEvent.dblClick(input);
+    await waitFor(
+      () => {
+        expect(
+          screen.getByText("A CQL library model is required.")
+        ).toBeInTheDocument();
+      },
+      { timeout: 5000 }
```

Comment:
Do we need a timeout, i believed waitFor can handle that.

Reply:
I'll double check. I think at one point I was having trouble with the waitfor's default timeout not being enough but I'll test this again and see if I can remove it.

Reply:
this was removed